### PR TITLE
Plot docs and convenience tools

### PR DESCRIPTION
**Pull request by keflavich**
Add plotting docs to address #101

### CODE REVIEW

File: docs/plotting.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+   ax.set_xlabel("Offset [arcmin]")
+
+See also the `tutorial <https://github.com/radio-astro-tools/tutorials/blob/6810376c0353f0bdf3be2b9b7231c388e886adba/PVDiagramPlotting.ipynb>`_
+
+
+Showing the extraction path
+---------------------------
+
+
+To show the extraction path, use the `shiw_on_axis` method:
```
Comment (e-koch) [inline docs/plotting.rst:41]:
```suggestion
To show the extraction path, use the :class:`~pvextractor.Path.show_on_axis` method:
```

File: docs/plotting.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+   ax = pl.subplot(111, projection=cube.wcs.celestial)
+   ax.imshow(cube.moment0(axis=0).value)
+
+   path.show_on_axis(ax, spacing=1)
+
+
+The `path` will be shown either as a set of lines if the path's width is zero
+or as a set of rectangles if the path has a finite width.
+
+The spacing should probably be set to the same spacing used for the PV
```
Comment (e-koch) [inline docs/plotting.rst:56]:
```suggestion
`spacing` should probably be set to the same spacing used for the PV
```

File: docs/programmatic.rst [hunk 3]
```diff
@@ -93,3 +93,17 @@ The returned slice is an Astropy :class:`~astropy.io.fits.PrimaryHDU` instance,
...
+
+
+Retrieving the Pixel Coordinates
+--------------------------------
+
+With a path in hand, you can recover the pixel coordinates used in the
+extraction::
+
+    >>> from spectral_cube import SpectralCube
+    >>> cube = SpectralCube.read('my_cube.fits')  # doctest: +SKIP
```
Comment (e-koch) [inline docs/programmatic.rst:105]: Is there a hosted spectral cube file that we can use in general so the example are tested?

File: pvextractor/geometry/path.py [hunk 4]
```diff
@@ -255,8 +255,54 @@ def sample_polygons(self, spacing, wcs=None):
...
-            patches.append(MPLPolygon(list(zip(poly.x, poly.y)), **kwargs))
-        return patches
+        if self.width is not None:
+            from matplotlib.patches import Polygon as MPLPolygon
+            patches = []
+            for poly in self.sample_polygons(spacing, wcs=wcs):
+                patches.append(MPLPolygon(list(zip(poly.x, poly.y)), **kwargs))
+            return patches
+        else:
+            raise ValueError("Use as_artist instead")
```
Comment (e-koch) [inline pvextractor/geometry/path.py:265]:
```suggestion
            raise ValueError("Use the `as_artist` function instead for paths with no defined width.")
```

Comment (e-koch) [reply pvextractor/geometry/path.py:265]: Hmm. I think this is ok, though I now see that `as_artist` is useful in both cases.

File: pvextractor/geometry/path.py [hunk 5]
```diff
@@ -255,8 +255,54 @@ def sample_polygons(self, spacing, wcs=None):
...
+        ----------
+        spacing : float
+            The spacing between sample points to plot, in pixels
+        wcs : astropy.wcs.WCS
+            A WCS instance to convert from celestial to pixel coordinates, or
+            None if pixel coordinates are to be used.
+        facecolor : string
+            The facecolor of the patches.  This will be ignored if lines are
+            plotted and defaults to none (transparent) for patches.
+        """
```
Comment (e-koch) [inline pvextractor/geometry/path.py:281]: Add kwargs getting passed to `matplotlib.collections.PatchCollection`.

File: pvextractor/geometry/path.py [hunk 6]
```diff
@@ -255,8 +255,54 @@ def sample_polygons(self, spacing, wcs=None):
...
+            patches = []
+            for poly in self.sample_polygons(spacing, wcs=wcs):
+                patches.append(MPLPolygon(list(zip(poly.x, poly.y)), **kwargs))
+            return patches
+        else:
+            raise ValueError("Use as_artist instead")
+
+    def as_artist(self, spacing, wcs=None, facecolor='none', **kwargs):
+        """
+        Display the path on the image
```
Comment (e-koch) [inline pvextractor/geometry/path.py:269]:
```suggestion
        Return the path as a matplotlib artist object that can be displayed on the image.
```
Maybe?

File: pvextractor/geometry/path.py [hunk 7]
```diff
@@ -255,8 +255,54 @@ def sample_polygons(self, spacing, wcs=None):
...
+        else:
+            from matplotlib.collections import PatchCollection
+            patches = self.to_patches(spacing, wcs=wcs, facecolor=facecolor,
+                                      **kwargs)
+            artist = PatchCollection(patches, facecolor=facecolor, **kwargs)
+        return artist
+
+    def show_on_axis(self, ax, spacing, **kwargs):
+        """
+        Add the Path as a set of patches on the specified axis
```
Comment (e-koch) [inline pvextractor/geometry/path.py:295]:
```suggestion
        Show the path as a set of patches on the specified matplotlib axis.
```

File: pvextractor/geometry/path.py [hunk 8]
```diff
@@ -255,8 +255,54 @@ def sample_polygons(self, spacing, wcs=None):
...
+        ax : matplotlib.axes._subplots.WCSAxesSubplot
+            An astropy WCSAxesSubplot axis to overplot onto
+        spacing : float
+            The spacing between sample points to plot, in pixels
+        """
+        artist = self.as_artist(spacing=spacing,
+                                wcs=ax.wcs,
+                                **kwargs)
+        ax.add_artist(artist)
+        return artist
```
Comment (e-koch) [inline pvextractor/geometry/path.py:308]: Does `artist` need to be returned? `as_artist` could be used instead if only the artist object is needed.

**Review by e-koch — CHANGES_REQUESTED** [review]

File: pvextractor/geometry/path.py (hunk 8 above)
Comment (keflavich) [reply pvextractor/geometry/path.py:308]: Doesn't need to, but matplotlib functions all return the artists, so I think we should

Comment (e-koch) [reply pvextractor/geometry/path.py:308]: 👍

File: docs/programmatic.rst (hunk 3 above)
Comment (keflavich) [reply docs/programmatic.rst:105]: Maybe we should use the L1448 cube from the example?

Comment (e-koch) [reply docs/programmatic.rst:105]: 👍

Comment (keflavich) [reply docs/programmatic.rst:105]: done

**Review by e-koch — APPROVED** [review]

**Pull request merged**